R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+  row_e0 <- c(round(e0, 1), e0_range)
+  row_q1_0 <- c(round(q1_0, 3), q1_0_range)
+  row_q5_0 <- c(round(q5_0, 3), q5_0_range)
+  e0_q5_0 <- data.frame(rbind(row_e0, row_q1_0, row_q5_0))
+  colnames(e0_q5_0) <- c("value", "range")
+  rownames(e0_q5_0) <- c("e0", "1q0", "5q0")
+  return(list(indices = e0_q5_0, life_table_corr = life_table_corr))
+  }
+
+
+#' Calculate masculinity index and maternal mortality
+#' from a mortAAR life table list
+#'
+#' The proportional relation between adult males and females (=
+#' Masculinity index) is interesting for a number of reasons: (1)
+#' it can point to basic problems in the datasets in that, say,
+#' one sex is grossly over- or underrepresented. (2) it may hint
+#' towards cultural reaons like sex-specific mobility.\cr
+#' Maternal mortality is a basic indicator for the health system of a
+#' given population. Recently, McFadden and Oxenham (2019) have
+#' provided a formula to calculate it from archaeological data.
+#'
+#' The Masculinity index (MI) is defined for juvenile and older
+#' individuals. Note that with a higher mortality rate of adult females,
+#' an MI < 100 does not necesssarily speak for an unbalanced MI in life.\cr
+#' Maternal mortality is calculated according to the formula
+#' provided by McFadden and Oxenham (2019). McFadden and Oxenham show
+#' that with modern data a very high correlation is achieved by only
+#' comparing the absolute numbers of the age group 20 to 24. This has the
+#' additional advantage that for this age group anthropological aging
+#' methods are reasonable exact.
+#'
+#' @param life_table an object of class mortaar_life_table_list.
+#' @param females an object of class mortaar_life_table for females
+#' @param males an object of class mortaar_life_table for males
+#'
+#' @return Output of masculinity index and maternal mortality.
+#' \itemize{
+#'   \item \bold{Masculinity index}.
+#'
+#'                    \eqn{MI = D>=15male * 100 / D>=15female}
+#'
+#'   \item \bold{Maternal mortality}.
+#'
+#'                    \eqn{333.33 * D20-24female / D20-24male - 76.07}
+#'}
+#'
+#' @examples
+#' # Calculate Masculinity index and maternal mortality from real life
+#' # dataset.
+#' library(magrittr)
+#' muen <- muensingen
+#' muen %>% dplyr::select(age) %>% dplyr::arrange(nchar(age), age) %>%
+#' unique()
+#' muen <- muen %>% dplyr::mutate( age = dplyr::case_when(
+#' .$age == ">50"    ~ "50-70", .$age == ">60"    ~ "60-70",
+#' .$age == "10+/-1" ~ "9-12", .$age == "10"     ~ "10-11",
+#' .$age == "7-8"    ~ "7-9", .$age == "9-10"   ~ "9-11",
+#' TRUE ~  .$age))
+#' muen <- muen %>% tidyr::separate(age, c("from", "to")) %>%
+#' transform(from = as.numeric(from), to = as.numeric(to))
+#' muen_prep <- muen %>% prep.life.table(group = "sex",
+#' agebeg = "from", ageend = "to", method = "Standard",
+#' agerange = "excluded")
+#' muen_result <- muen_prep %>% life.table()
+#' lt.sexrelation(muen_result,"female","male")
+#'
```

Comment:
This example code is a bit overwhelming. Too specific and too much stuff that is not related to `lt.sexrelation`. Maybe it's better to create a new example dataset that can be used directly with this function. Without the tedious preparation.

This also applies for the other functions.

Reply:
Yes, you are right. I noticed it already during setting up the examples. I will look for something appropriate ...